relayer/relays/beacon/header/syncer/scale/beacon_scale.go

```diff
@@ -294,3 +295,26 @@ type BeaconState struct {
 	CompactBeaconState
 	BlockRoot types.H256
 }
+
+type VersionedExecutionPayloadHeader struct {
+	Capella *ExecutionPayloadHeaderCapella
+	Deneb   *ExecutionPayloadHeaderDeneb
+}
+
+var (
+	ErrEncodeVersionedExecutionPayloadHeader = errors.New("error scale encode VersionedExecutionPayloadHeader")
+)
+
+func (v VersionedExecutionPayloadHeader) Encode(encoder scale.Encoder) error {
+	var err error
+	if v.Capella != nil {
+		encoder.PushByte(0)
+		err = encoder.Encode(v.Capella)
+	} else if v.Deneb != nil {
+		encoder.PushByte(1)
```

Comment:
Why is this byte prefix needed?

Reply:
It's required to match the codec encoding of `VersionedExecutionPayloadHeader` in the beacon light client.

Reply:
Can't we just use the fork version to determine which version of the header to use?

Reply:
Yes, we can. But the point here is to use a single extrinsic to provide multiple version support.